crm/modules/user/user.inc.php

```diff
@@ -866,6 +866,12 @@ function command_set_password () {
     global $db_connect;
     global $esc_post;
     
+    // Check permissions
+    if (!user_access('user_edit')) {
+        error_register('Current user does not have permission: user_edit');
+        return crm_url('permissions');
```

Comment:
The above would forbid a user to edit their own password. There needs to be another conditional to check if user is setting/changing their own password.

Reply:
just spotted that myself, already working on it :)